scenario-tests/run_scenario_tests.sh

```diff
@@ -40,30 +51,57 @@ while [ $# -gt 0 ]
   do case $1 in
     --exclude=*)            TEST_EXCLUDE_REGEX=${1##--exclude=} ;;
     --include=*)            TEST_INCLUDE_REGEX=${1##--include=} ;;
+    --docker-python2.7)     USE_DOCKER=1;DOCKER_PYTHON=2.7;;
+    --docker-python3.5)     USE_DOCKER=1;DOCKER_PYTHON=3.5;;
+    --docker-python3.6)     USE_DOCKER=1;DOCKER_PYTHON=3.6;;
     --help)                 print_help ;;
-    *) GLC_EGG_UNDER_TEST=$1 ;;
+    *) TC_WHEEL_UNDER_TEST=$1 ;;
   esac
   shift
 done
 
-if [[ $GLC_EGG_UNDER_TEST == "" ]]; then
+if [[ $TC_WHEEL_UNDER_TEST == "" ]]; then
         # No URL/package name specified on the command line
         # use environment variables
-        OS="`uname -s`"
-        if [[ $OS == 'Linux' ]]; then
-                GLC_EGG_UNDER_TEST=$GLC_LINUX_EGG_URL
-        elif [[ $OS == 'Darwin' ]]; then
-                GLC_EGG_UNDER_TEST=$GLC_MAC_EGG_URL
-        elif [[ `uname -o` == 'Msys' ]]; then
-                GLC_EGG_UNDER_TEST=$GLC_WINDOWS_EGG_URL
-        fi
+        TC_WHEEL_UNDER_TEST=$TC_WHEEL_PATH
 fi
 
-if [[ $GLC_EGG_UNDER_TEST == "" ]]; then
+if [[ $TC_WHEEL_UNDER_TEST == "" ]]; then
         # Still no egg set -- show help
         print_help
 fi
 
+# If we are going to run in Docker,
+# send this command into Docker and bail out of here when done.
+if [[ -n "${USE_DOCKER}" ]]; then
+  # create the build and test images
+  # (this should ideally be a no-op if the image exists and is current)
+  $WORKSPACE/scripts/create_docker_images.sh
+
+  # Run the tests inside Docker
+  if [[ "${DOCKER_PYTHON}" == "2.7" ]]; then
+    docker run --rm -m=4g \
+      --mount type=bind,source=$WORKSPACE,target=/build,consistency=delegated \
+      -e "VIRTUALENV=virtualenv --python=python${DOCKER_PYTHON}" \
+      turicreate/build-image-14.04:${TC_BUILD_IMAGE_VERSION} \
+      /build/scenario-tests/run_scenario_tests.sh $TC_WHEEL_UNDER_TEST
+  elif [[ "${DOCKER_PYTHON}" == "3.5" ]]; then
+    docker run --rm -m=4g \
```

Comment:
The body of this `elif` block seem identical to the body of the `if` statement above it. If that is indeed the case, I think you should combine them, i.e the `if` test should check if it's 2.7 or 3.5.